{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fee2ed17",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import json\n",
    "import pandas as pd\n",
    "import requests\n",
    "from pandas.core.frame import DataFrame\n",
    "\n",
    "import numpy as np\n",
    "from sklearn.preprocessing import FunctionTransformer, StandardScaler\n",
    "from sklearn.pipeline import Pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "425ee33e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "nr. reads seqeunced",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "host age in years",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "BMI",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "recent antibiotics use",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "QC status",
         "rawType": "float64",
         "type": "float"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "d1031fa0-9261-40d1-8139-7bbfc484efd8",
       "rows": [
        [
         "count",
         "13.0",
         "56.0",
         "13.0",
         "0.0",
         "219.0"
        ],
        [
         "mean",
         "34847.38461538462",
         "64.05357142857143",
         "21.299230769230768",
         null,
         "0.9680365296803652"
        ],
        [
         "std",
         "58549.152454068404",
         "19.156218684430232",
         "9.882346601843834",
         null,
         "0.17630581890206298"
        ],
        [
         "min",
         "1.0",
         "0.0",
         "0.0",
         null,
         "0.0"
        ],
        [
         "25%",
         "12989.0",
         "59.0",
         "21.3",
         null,
         "1.0"
        ],
        [
         "50%",
         "23205.0",
         "67.0",
         "24.41",
         null,
         "1.0"
        ],
        [
         "75%",
         "25836.0",
         "78.0",
         "26.09",
         null,
         "1.0"
        ],
        [
         "max",
         "227161.0",
         "85.0",
         "30.85",
         null,
         "1.0"
        ]
       ],
       "shape": {
        "columns": 5,
        "rows": 8
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>nr. reads seqeunced</th>\n",
       "      <th>host age in years</th>\n",
       "      <th>BMI</th>\n",
       "      <th>recent antibiotics use</th>\n",
       "      <th>QC status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>13.000000</td>\n",
       "      <td>56.000000</td>\n",
       "      <td>13.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>219.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>34847.384615</td>\n",
       "      <td>64.053571</td>\n",
       "      <td>21.299231</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.968037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>58549.152454</td>\n",
       "      <td>19.156219</td>\n",
       "      <td>9.882347</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.176306</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>12989.000000</td>\n",
       "      <td>59.000000</td>\n",
       "      <td>21.300000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>23205.000000</td>\n",
       "      <td>67.000000</td>\n",
       "      <td>24.410000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>25836.000000</td>\n",
       "      <td>78.000000</td>\n",
       "      <td>26.090000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>227161.000000</td>\n",
       "      <td>85.000000</td>\n",
       "      <td>30.850000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       nr. reads seqeunced  host age in years        BMI  \\\n",
       "count            13.000000          56.000000  13.000000   \n",
       "mean          34847.384615          64.053571  21.299231   \n",
       "std           58549.152454          19.156219   9.882347   \n",
       "min               1.000000           0.000000   0.000000   \n",
       "25%           12989.000000          59.000000  21.300000   \n",
       "50%           23205.000000          67.000000  24.410000   \n",
       "75%           25836.000000          78.000000  26.090000   \n",
       "max          227161.000000          85.000000  30.850000   \n",
       "\n",
       "       recent antibiotics use   QC status  \n",
       "count                     0.0  219.000000  \n",
       "mean                      NaN    0.968037  \n",
       "std                       NaN    0.176306  \n",
       "min                       NaN    0.000000  \n",
       "25%                       NaN    1.000000  \n",
       "50%                       NaN    1.000000  \n",
       "75%                       NaN    1.000000  \n",
       "max                       NaN    1.000000  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_pr1 = pd.read_csv('Data/all_runs_in_project_PRJNA485797.tsv', sep=\"\t\" , index_col=False)\n",
    "df_pr1.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "19ca63a9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "int64",
         "type": "integer"
        },
        {
         "name": "project ID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "run ID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "sample ID",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "experiment type",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "nr. reads seqeunced",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "instrument model",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "country",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "collection  date",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "gender",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "host age in years",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "diet",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "BMI",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "recent antibiotics use",
         "rawType": "float64",
         "type": "float"
        },
        {
         "name": "QC status",
         "rawType": "int64",
         "type": "integer"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "575cb25c-49f9-4069-870f-6076f79a96ed",
       "rows": [
        [
         "0",
         "PRJEB11419",
         "ERR1090510",
         "ERS940223",
         "Amplicon",
         "26067.0",
         "Illumina MiSeq",
         "United States of America",
         "2015-03-18",
         "Male",
         "31.0",
         "Omnivore",
         "30.11",
         null,
         "1"
        ],
        [
         "2",
         "PRJEB11419",
         "ERR1160800",
         "ERS994317",
         "Amplicon",
         "227161.0",
         "Illumina HiSeq 2000",
         "United States of America",
         "2015-10-23",
         "Female",
         "53.0",
         "Omnivore",
         "26.09",
         null,
         "1"
        ],
        [
         "7",
         "PRJEB11419",
         "ERR1843463",
         "ERS1561262",
         "Amplicon",
         "23205.0",
         "Illumina MiSeq",
         "United States of America",
         "2015-05-18",
         "Male",
         "0.0",
         "Omnivore",
         "24.41",
         null,
         "1"
        ],
        [
         "8",
         "PRJEB11419",
         "ERR1844528",
         "ERS1561262",
         "Amplicon",
         "23303.0",
         "Illumina MiSeq",
         "United States of America",
         "2015-05-18",
         "Male",
         "0.0",
         "Omnivore",
         "24.41",
         null,
         "1"
        ],
        [
         "10",
         "PRJEB11419",
         "ERR2091834",
         "ERS1865177",
         "Amplicon",
         "25836.0",
         "Illumina MiSeq",
         "United Kingdom",
         "2016-12-07",
         "Male",
         "56.0",
         "Omnivore",
         "23.72",
         null,
         "1"
        ]
       ],
       "shape": {
        "columns": 14,
        "rows": 5
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>project ID</th>\n",
       "      <th>run ID</th>\n",
       "      <th>sample ID</th>\n",
       "      <th>experiment type</th>\n",
       "      <th>nr. reads seqeunced</th>\n",
       "      <th>instrument model</th>\n",
       "      <th>country</th>\n",
       "      <th>collection  date</th>\n",
       "      <th>gender</th>\n",
       "      <th>host age in years</th>\n",
       "      <th>diet</th>\n",
       "      <th>BMI</th>\n",
       "      <th>recent antibiotics use</th>\n",
       "      <th>QC status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PRJEB11419</td>\n",
       "      <td>ERR1090510</td>\n",
       "      <td>ERS940223</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>26067.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>2015-03-18</td>\n",
       "      <td>Male</td>\n",
       "      <td>31.0</td>\n",
       "      <td>Omnivore</td>\n",
       "      <td>30.11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PRJEB11419</td>\n",
       "      <td>ERR1160800</td>\n",
       "      <td>ERS994317</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>227161.0</td>\n",
       "      <td>Illumina HiSeq 2000</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>2015-10-23</td>\n",
       "      <td>Female</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Omnivore</td>\n",
       "      <td>26.09</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>PRJEB11419</td>\n",
       "      <td>ERR1843463</td>\n",
       "      <td>ERS1561262</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>23205.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>2015-05-18</td>\n",
       "      <td>Male</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Omnivore</td>\n",
       "      <td>24.41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>PRJEB11419</td>\n",
       "      <td>ERR1844528</td>\n",
       "      <td>ERS1561262</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>23303.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>United States of America</td>\n",
       "      <td>2015-05-18</td>\n",
       "      <td>Male</td>\n",
       "      <td>0.0</td>\n",
       "      <td>Omnivore</td>\n",
       "      <td>24.41</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>PRJEB11419</td>\n",
       "      <td>ERR2091834</td>\n",
       "      <td>ERS1865177</td>\n",
       "      <td>Amplicon</td>\n",
       "      <td>25836.0</td>\n",
       "      <td>Illumina MiSeq</td>\n",
       "      <td>United Kingdom</td>\n",
       "      <td>2016-12-07</td>\n",
       "      <td>Male</td>\n",
       "      <td>56.0</td>\n",
       "      <td>Omnivore</td>\n",
       "      <td>23.72</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    project ID      run ID   sample ID experiment type  nr. reads seqeunced  \\\n",
       "0   PRJEB11419  ERR1090510   ERS940223        Amplicon              26067.0   \n",
       "2   PRJEB11419  ERR1160800   ERS994317        Amplicon             227161.0   \n",
       "7   PRJEB11419  ERR1843463  ERS1561262        Amplicon              23205.0   \n",
       "8   PRJEB11419  ERR1844528  ERS1561262        Amplicon              23303.0   \n",
       "10  PRJEB11419  ERR2091834  ERS1865177        Amplicon              25836.0   \n",
       "\n",
       "       instrument model                   country collection  date  gender  \\\n",
       "0        Illumina MiSeq  United States of America       2015-03-18    Male   \n",
       "2   Illumina HiSeq 2000  United States of America       2015-10-23  Female   \n",
       "7        Illumina MiSeq  United States of America       2015-05-18    Male   \n",
       "8        Illumina MiSeq  United States of America       2015-05-18    Male   \n",
       "10       Illumina MiSeq            United Kingdom       2016-12-07    Male   \n",
       "\n",
       "    host age in years      diet    BMI  recent antibiotics use  QC status  \n",
       "0                31.0  Omnivore  30.11                     NaN          1  \n",
       "2                53.0  Omnivore  26.09                     NaN          1  \n",
       "7                 0.0  Omnivore  24.41                     NaN          1  \n",
       "8                 0.0  Omnivore  24.41                     NaN          1  \n",
       "10               56.0  Omnivore  23.72                     NaN          1  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_QC1 = df_pr1.loc[df_pr1['QC status'] == 1.0]\n",
    "#df_phen_health = df_QC1.loc[df_QC1['associated phenotype'] == 'D006262'] #healthy\n",
    "df_QC1.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "2bc96232",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count            212\n",
       "unique           212\n",
       "top       ERR1090510\n",
       "freq               1\n",
       "Name: run ID, dtype: object"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_runs = df_QC1['run ID']\n",
    "df_runs.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "9fd11e49",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_only_runs = pd.DataFrame({'Runs': df_runs.values})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9c839582",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.microsoft.datawrangler.viewer.v0+json": {
       "columns": [
        {
         "name": "index",
         "rawType": "object",
         "type": "string"
        },
        {
         "name": "Runs",
         "rawType": "object",
         "type": "unknown"
        }
       ],
       "conversionMethod": "pd.DataFrame",
       "ref": "115ac888-9952-45e6-b863-d64e8d78a318",
       "rows": [
        [
         "count",
         "212"
        ],
        [
         "unique",
         "212"
        ],
        [
         "top",
         "ERR1090510"
        ],
        [
         "freq",
         "1"
        ]
       ],
       "shape": {
        "columns": 1,
        "rows": 4
       }
      },
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Runs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>unique</th>\n",
       "      <td>212</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>top</th>\n",
       "      <td>ERR1090510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>freq</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Runs\n",
       "count          212\n",
       "unique         212\n",
       "top     ERR1090510\n",
       "freq             1"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_only_runs.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4f4290a9",
   "metadata": {},
   "outputs": [],
   "source": [
    "all_flattened_runs = []\n",
    "identifier = 'Healthy'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e11b94f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825857', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66356, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825862', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66361, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825873', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66372, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825884', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66383, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825895', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66394, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825896', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66395, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825897', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66396, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825898', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66397, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825899', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66398, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825900', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66399, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825901', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66400, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825902', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66401, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825903', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66402, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825904', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66403, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825905', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66404, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825906', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66405, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825907', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66406, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825908', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66407, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825909', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66408, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825910', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66409, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825911', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66410, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825912', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66411, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825913', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66412, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825914', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66413, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825915', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66414, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825916', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66415, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825917', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66416, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825918', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66417, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825919', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66418, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825920', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66419, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825921', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66420, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825922', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66421, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825923', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66422, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825924', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66423, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825925', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66424, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825926', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66425, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825927', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66426, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825928', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66427, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825930', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66429, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825941', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66440, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825952', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66451, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825963', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66462, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825970', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66469, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825971', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66470, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825972', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66471, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825973', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66472, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825974', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66473, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825975', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66474, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825976', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66475, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825977', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66476, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825978', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66477, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825979', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66478, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825980', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66479, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825981', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66480, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825982', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66481, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825983', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66482, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825984', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66483, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825985', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66484, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825986', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66485, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825987', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66486, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825988', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66487, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825989', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66488, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n",
      "Error processing run {'project_id': 'PRJNA633959', 'original_sample_description': None, 'run_id': 'SRR11825990', 'experiment_type': 'Amplicon', 'instrument_model': 'Illumina MiSeq', 'nr_reads_sequenced': None, 'host_age': None, 'sex': None, 'BMI': None, 'country': 'China', 'longitude': None, 'latitude': None, 'loaded_uid': 66489, 'QCStatus': 1, 'QCMessage': 'p-trim-left-f 0;p-trim-left-r 19;p-trunc-len-f 290;p-trunc-len-r 260', 'Original_Project_description': 'With the development of the next-generation sequencing techniques, the higher sequencing depth and coverage made it possible to decipher those unknown key functional taxa in Chinese AD patients. In our present study, the AD associated fecal microbiota were analyzed from AD patients and matched healthy controls from Lishui by 16S rRNA gene high-throughput sequencing platform, which will provide novel targets for the non-invasive diagnosis and personalized treatment of AD as well as the development of tailor-made probiotics designed for Chinese AD patients.\\r\\n'}: \"['taxon_rank_level', 'scientific_name'] not found in axis\"\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[8], line 5\u001b[0m\n\u001b[0;32m      3\u001b[0m query \u001b[38;5;241m=\u001b[39m {\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrun_id\u001b[39m\u001b[38;5;124m\"\u001b[39m:run_id}  \n\u001b[0;32m      4\u001b[0m url \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mhttps://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID\u001b[39m\u001b[38;5;124m'\u001b[39m\n\u001b[1;32m----> 5\u001b[0m data \u001b[38;5;241m=\u001b[39m requests\u001b[38;5;241m.\u001b[39mpost(url, data\u001b[38;5;241m=\u001b[39mjson\u001b[38;5;241m.\u001b[39mdumps(query))\u001b[38;5;241m.\u001b[39mjson()\n\u001b[0;32m      7\u001b[0m \u001b[38;5;66;03m## --get run List\u001b[39;00m\n\u001b[0;32m      8\u001b[0m run \u001b[38;5;241m=\u001b[39m data\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mrun\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\requests\\api.py:115\u001b[0m, in \u001b[0;36mpost\u001b[1;34m(url, data, json, **kwargs)\u001b[0m\n\u001b[0;32m    103\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mpost\u001b[39m(url, data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, json\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs):\n\u001b[0;32m    104\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124mr\u001b[39m\u001b[38;5;124;03m\"\"\"Sends a POST request.\u001b[39;00m\n\u001b[0;32m    105\u001b[0m \n\u001b[0;32m    106\u001b[0m \u001b[38;5;124;03m    :param url: URL for the new :class:`Request` object.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    112\u001b[0m \u001b[38;5;124;03m    :rtype: requests.Response\u001b[39;00m\n\u001b[0;32m    113\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> 115\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m request(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mpost\u001b[39m\u001b[38;5;124m\"\u001b[39m, url, data\u001b[38;5;241m=\u001b[39mdata, json\u001b[38;5;241m=\u001b[39mjson, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\requests\\api.py:59\u001b[0m, in \u001b[0;36mrequest\u001b[1;34m(method, url, **kwargs)\u001b[0m\n\u001b[0;32m     55\u001b[0m \u001b[38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we\u001b[39;00m\n\u001b[0;32m     56\u001b[0m \u001b[38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some\u001b[39;00m\n\u001b[0;32m     57\u001b[0m \u001b[38;5;66;03m# cases, and look like a memory leak in others.\u001b[39;00m\n\u001b[0;32m     58\u001b[0m \u001b[38;5;28;01mwith\u001b[39;00m sessions\u001b[38;5;241m.\u001b[39mSession() \u001b[38;5;28;01mas\u001b[39;00m session:\n\u001b[1;32m---> 59\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m session\u001b[38;5;241m.\u001b[39mrequest(method\u001b[38;5;241m=\u001b[39mmethod, url\u001b[38;5;241m=\u001b[39murl, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:589\u001b[0m, in \u001b[0;36mSession.request\u001b[1;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)\u001b[0m\n\u001b[0;32m    584\u001b[0m send_kwargs \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m    585\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtimeout\u001b[39m\u001b[38;5;124m\"\u001b[39m: timeout,\n\u001b[0;32m    586\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mallow_redirects\u001b[39m\u001b[38;5;124m\"\u001b[39m: allow_redirects,\n\u001b[0;32m    587\u001b[0m }\n\u001b[0;32m    588\u001b[0m send_kwargs\u001b[38;5;241m.\u001b[39mupdate(settings)\n\u001b[1;32m--> 589\u001b[0m resp \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msend(prep, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39msend_kwargs)\n\u001b[0;32m    591\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m resp\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\requests\\sessions.py:703\u001b[0m, in \u001b[0;36mSession.send\u001b[1;34m(self, request, **kwargs)\u001b[0m\n\u001b[0;32m    700\u001b[0m start \u001b[38;5;241m=\u001b[39m preferred_clock()\n\u001b[0;32m    702\u001b[0m \u001b[38;5;66;03m# Send the request\u001b[39;00m\n\u001b[1;32m--> 703\u001b[0m r \u001b[38;5;241m=\u001b[39m adapter\u001b[38;5;241m.\u001b[39msend(request, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwargs)\n\u001b[0;32m    705\u001b[0m \u001b[38;5;66;03m# Total elapsed time of the request (approximately)\u001b[39;00m\n\u001b[0;32m    706\u001b[0m elapsed \u001b[38;5;241m=\u001b[39m preferred_clock() \u001b[38;5;241m-\u001b[39m start\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\requests\\adapters.py:486\u001b[0m, in \u001b[0;36mHTTPAdapter.send\u001b[1;34m(self, request, stream, timeout, verify, cert, proxies)\u001b[0m\n\u001b[0;32m    483\u001b[0m     timeout \u001b[38;5;241m=\u001b[39m TimeoutSauce(connect\u001b[38;5;241m=\u001b[39mtimeout, read\u001b[38;5;241m=\u001b[39mtimeout)\n\u001b[0;32m    485\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 486\u001b[0m     resp \u001b[38;5;241m=\u001b[39m conn\u001b[38;5;241m.\u001b[39murlopen(\n\u001b[0;32m    487\u001b[0m         method\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mmethod,\n\u001b[0;32m    488\u001b[0m         url\u001b[38;5;241m=\u001b[39murl,\n\u001b[0;32m    489\u001b[0m         body\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mbody,\n\u001b[0;32m    490\u001b[0m         headers\u001b[38;5;241m=\u001b[39mrequest\u001b[38;5;241m.\u001b[39mheaders,\n\u001b[0;32m    491\u001b[0m         redirect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    492\u001b[0m         assert_same_host\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    493\u001b[0m         preload_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    494\u001b[0m         decode_content\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    495\u001b[0m         retries\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmax_retries,\n\u001b[0;32m    496\u001b[0m         timeout\u001b[38;5;241m=\u001b[39mtimeout,\n\u001b[0;32m    497\u001b[0m         chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    498\u001b[0m     )\n\u001b[0;32m    500\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (ProtocolError, \u001b[38;5;167;01mOSError\u001b[39;00m) \u001b[38;5;28;01mas\u001b[39;00m err:\n\u001b[0;32m    501\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mConnectionError\u001b[39;00m(err, request\u001b[38;5;241m=\u001b[39mrequest)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:714\u001b[0m, in \u001b[0;36mHTTPConnectionPool.urlopen\u001b[1;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, **response_kw)\u001b[0m\n\u001b[0;32m    711\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_prepare_proxy(conn)\n\u001b[0;32m    713\u001b[0m \u001b[38;5;66;03m# Make the request on the httplib connection object.\u001b[39;00m\n\u001b[1;32m--> 714\u001b[0m httplib_response \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_make_request(\n\u001b[0;32m    715\u001b[0m     conn,\n\u001b[0;32m    716\u001b[0m     method,\n\u001b[0;32m    717\u001b[0m     url,\n\u001b[0;32m    718\u001b[0m     timeout\u001b[38;5;241m=\u001b[39mtimeout_obj,\n\u001b[0;32m    719\u001b[0m     body\u001b[38;5;241m=\u001b[39mbody,\n\u001b[0;32m    720\u001b[0m     headers\u001b[38;5;241m=\u001b[39mheaders,\n\u001b[0;32m    721\u001b[0m     chunked\u001b[38;5;241m=\u001b[39mchunked,\n\u001b[0;32m    722\u001b[0m )\n\u001b[0;32m    724\u001b[0m \u001b[38;5;66;03m# If we're going to release the connection in ``finally:``, then\u001b[39;00m\n\u001b[0;32m    725\u001b[0m \u001b[38;5;66;03m# the response doesn't need to know about the connection. Otherwise\u001b[39;00m\n\u001b[0;32m    726\u001b[0m \u001b[38;5;66;03m# it will also try to release it and we'll have a double-release\u001b[39;00m\n\u001b[0;32m    727\u001b[0m \u001b[38;5;66;03m# mess.\u001b[39;00m\n\u001b[0;32m    728\u001b[0m response_conn \u001b[38;5;241m=\u001b[39m conn \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m release_conn \u001b[38;5;28;01melse\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:403\u001b[0m, in \u001b[0;36mHTTPConnectionPool._make_request\u001b[1;34m(self, conn, method, url, timeout, chunked, **httplib_request_kw)\u001b[0m\n\u001b[0;32m    401\u001b[0m \u001b[38;5;66;03m# Trigger any extra validation we need to do.\u001b[39;00m\n\u001b[0;32m    402\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 403\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_validate_conn(conn)\n\u001b[0;32m    404\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (SocketTimeout, BaseSSLError) \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    405\u001b[0m     \u001b[38;5;66;03m# Py2 raises this as a BaseSSLError, Py3 raises it as socket timeout.\u001b[39;00m\n\u001b[0;32m    406\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_raise_timeout(err\u001b[38;5;241m=\u001b[39me, url\u001b[38;5;241m=\u001b[39murl, timeout_value\u001b[38;5;241m=\u001b[39mconn\u001b[38;5;241m.\u001b[39mtimeout)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\connectionpool.py:1053\u001b[0m, in \u001b[0;36mHTTPSConnectionPool._validate_conn\u001b[1;34m(self, conn)\u001b[0m\n\u001b[0;32m   1051\u001b[0m \u001b[38;5;66;03m# Force connect early to allow us to validate the connection.\u001b[39;00m\n\u001b[0;32m   1052\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mgetattr\u001b[39m(conn, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msock\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m):  \u001b[38;5;66;03m# AppEngine might not have  `.sock`\u001b[39;00m\n\u001b[1;32m-> 1053\u001b[0m     conn\u001b[38;5;241m.\u001b[39mconnect()\n\u001b[0;32m   1055\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m conn\u001b[38;5;241m.\u001b[39mis_verified:\n\u001b[0;32m   1056\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[0;32m   1057\u001b[0m         (\n\u001b[0;32m   1058\u001b[0m             \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnverified HTTPS request is being made to host \u001b[39m\u001b[38;5;124m'\u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m'\u001b[39m\u001b[38;5;124m. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1063\u001b[0m         InsecureRequestWarning,\n\u001b[0;32m   1064\u001b[0m     )\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\connection.py:419\u001b[0m, in \u001b[0;36mHTTPSConnection.connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    410\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[0;32m    411\u001b[0m     \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_certs\n\u001b[0;32m    412\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_cert_dir\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    415\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(context, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mload_default_certs\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m    416\u001b[0m ):\n\u001b[0;32m    417\u001b[0m     context\u001b[38;5;241m.\u001b[39mload_default_certs()\n\u001b[1;32m--> 419\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock \u001b[38;5;241m=\u001b[39m ssl_wrap_socket(\n\u001b[0;32m    420\u001b[0m     sock\u001b[38;5;241m=\u001b[39mconn,\n\u001b[0;32m    421\u001b[0m     keyfile\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mkey_file,\n\u001b[0;32m    422\u001b[0m     certfile\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mcert_file,\n\u001b[0;32m    423\u001b[0m     key_password\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mkey_password,\n\u001b[0;32m    424\u001b[0m     ca_certs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_certs,\n\u001b[0;32m    425\u001b[0m     ca_cert_dir\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_cert_dir,\n\u001b[0;32m    426\u001b[0m     ca_cert_data\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mca_cert_data,\n\u001b[0;32m    427\u001b[0m     server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname,\n\u001b[0;32m    428\u001b[0m     ssl_context\u001b[38;5;241m=\u001b[39mcontext,\n\u001b[0;32m    429\u001b[0m     tls_in_tls\u001b[38;5;241m=\u001b[39mtls_in_tls,\n\u001b[0;32m    430\u001b[0m )\n\u001b[0;32m    432\u001b[0m \u001b[38;5;66;03m# If we're using all defaults and the connection\u001b[39;00m\n\u001b[0;32m    433\u001b[0m \u001b[38;5;66;03m# is TLSv1 or TLSv1.1 we throw a DeprecationWarning\u001b[39;00m\n\u001b[0;32m    434\u001b[0m \u001b[38;5;66;03m# for the host.\u001b[39;00m\n\u001b[0;32m    435\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m (\n\u001b[0;32m    436\u001b[0m     default_ssl_context\n\u001b[0;32m    437\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mssl_version \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m    438\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mhasattr\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mversion\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m    439\u001b[0m     \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msock\u001b[38;5;241m.\u001b[39mversion() \u001b[38;5;129;01min\u001b[39;00m {\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTLSv1\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mTLSv1.1\u001b[39m\u001b[38;5;124m\"\u001b[39m}\n\u001b[0;32m    440\u001b[0m ):\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\util\\ssl_.py:449\u001b[0m, in \u001b[0;36mssl_wrap_socket\u001b[1;34m(sock, keyfile, certfile, cert_reqs, ca_certs, server_hostname, ssl_version, ciphers, ssl_context, ca_cert_dir, key_password, ca_cert_data, tls_in_tls)\u001b[0m\n\u001b[0;32m    437\u001b[0m     warnings\u001b[38;5;241m.\u001b[39mwarn(\n\u001b[0;32m    438\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAn HTTPS request has been made, but the SNI (Server Name \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[0;32m    439\u001b[0m         \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mIndication) extension to TLS is not available on this platform. \u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    445\u001b[0m         SNIMissingWarning,\n\u001b[0;32m    446\u001b[0m     )\n\u001b[0;32m    448\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m send_sni:\n\u001b[1;32m--> 449\u001b[0m     ssl_sock \u001b[38;5;241m=\u001b[39m _ssl_wrap_socket_impl(\n\u001b[0;32m    450\u001b[0m         sock, context, tls_in_tls, server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname\n\u001b[0;32m    451\u001b[0m     )\n\u001b[0;32m    452\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    453\u001b[0m     ssl_sock \u001b[38;5;241m=\u001b[39m _ssl_wrap_socket_impl(sock, context, tls_in_tls)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\site-packages\\urllib3\\util\\ssl_.py:493\u001b[0m, in \u001b[0;36m_ssl_wrap_socket_impl\u001b[1;34m(sock, ssl_context, tls_in_tls, server_hostname)\u001b[0m\n\u001b[0;32m    490\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m SSLTransport(sock, ssl_context, server_hostname)\n\u001b[0;32m    492\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m server_hostname:\n\u001b[1;32m--> 493\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ssl_context\u001b[38;5;241m.\u001b[39mwrap_socket(sock, server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname)\n\u001b[0;32m    494\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    495\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m ssl_context\u001b[38;5;241m.\u001b[39mwrap_socket(sock)\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\ssl.py:517\u001b[0m, in \u001b[0;36mSSLContext.wrap_socket\u001b[1;34m(self, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, session)\u001b[0m\n\u001b[0;32m    511\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mwrap_socket\u001b[39m(\u001b[38;5;28mself\u001b[39m, sock, server_side\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m,\n\u001b[0;32m    512\u001b[0m                 do_handshake_on_connect\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    513\u001b[0m                 suppress_ragged_eofs\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mTrue\u001b[39;00m,\n\u001b[0;32m    514\u001b[0m                 server_hostname\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m, session\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mNone\u001b[39;00m):\n\u001b[0;32m    515\u001b[0m     \u001b[38;5;66;03m# SSLSocket class handles server_hostname encoding before it calls\u001b[39;00m\n\u001b[0;32m    516\u001b[0m     \u001b[38;5;66;03m# ctx._wrap_socket()\u001b[39;00m\n\u001b[1;32m--> 517\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msslsocket_class\u001b[38;5;241m.\u001b[39m_create(\n\u001b[0;32m    518\u001b[0m         sock\u001b[38;5;241m=\u001b[39msock,\n\u001b[0;32m    519\u001b[0m         server_side\u001b[38;5;241m=\u001b[39mserver_side,\n\u001b[0;32m    520\u001b[0m         do_handshake_on_connect\u001b[38;5;241m=\u001b[39mdo_handshake_on_connect,\n\u001b[0;32m    521\u001b[0m         suppress_ragged_eofs\u001b[38;5;241m=\u001b[39msuppress_ragged_eofs,\n\u001b[0;32m    522\u001b[0m         server_hostname\u001b[38;5;241m=\u001b[39mserver_hostname,\n\u001b[0;32m    523\u001b[0m         context\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m,\n\u001b[0;32m    524\u001b[0m         session\u001b[38;5;241m=\u001b[39msession\n\u001b[0;32m    525\u001b[0m     )\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\ssl.py:1075\u001b[0m, in \u001b[0;36mSSLSocket._create\u001b[1;34m(cls, sock, server_side, do_handshake_on_connect, suppress_ragged_eofs, server_hostname, context, session)\u001b[0m\n\u001b[0;32m   1072\u001b[0m         \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m:\n\u001b[0;32m   1073\u001b[0m             \u001b[38;5;66;03m# non-blocking\u001b[39;00m\n\u001b[0;32m   1074\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdo_handshake_on_connect should not be specified for non-blocking sockets\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[1;32m-> 1075\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[0;32m   1076\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m (\u001b[38;5;167;01mOSError\u001b[39;00m, \u001b[38;5;167;01mValueError\u001b[39;00m):\n\u001b[0;32m   1077\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mclose()\n",
      "File \u001b[1;32mc:\\Users\\Daria\\anaconda3\\Lib\\ssl.py:1346\u001b[0m, in \u001b[0;36mSSLSocket.do_handshake\u001b[1;34m(self, block)\u001b[0m\n\u001b[0;32m   1344\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m timeout \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m0.0\u001b[39m \u001b[38;5;129;01mand\u001b[39;00m block:\n\u001b[0;32m   1345\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(\u001b[38;5;28;01mNone\u001b[39;00m)\n\u001b[1;32m-> 1346\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_sslobj\u001b[38;5;241m.\u001b[39mdo_handshake()\n\u001b[0;32m   1347\u001b[0m \u001b[38;5;28;01mfinally\u001b[39;00m:\n\u001b[0;32m   1348\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39msettimeout(timeout)\n",
      "\u001b[1;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "for index, run_id in df_only_runs[\"Runs\"].items():\n",
    "    try:\n",
    "        query = {\"run_id\":run_id}  \n",
    "        url = 'https://gmrepo.humangut.info/api/getFullTaxonomicProfileByRunID'\n",
    "        data = requests.post(url, data=json.dumps(query)).json()\n",
    "\n",
    "        ## --get run List\n",
    "        run = data.get(\"run\")\n",
    "\n",
    "        ## --get DataFrames\n",
    "        species = DataFrame(data.get(\"species\"))\n",
    "        species = species.drop(columns = ['taxon_rank_level', 'scientific_name']) #drop useless info\n",
    "\n",
    "        #species = species[species['ncbi_taxon_id'] != -1]   #drop missing taxons\n",
    "        flat = species.set_index('ncbi_taxon_id')['relative_abundance'].T\n",
    "        flat = flat.rename_axis(None).to_frame().T\n",
    "        flat.insert(0, 'Run', run_id)\n",
    "        all_flattened_runs.append(flat)\n",
    "    except Exception as e:\n",
    "        print(f\"Error processing run {run}: {e}\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40676138",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Daria\\AppData\\Local\\Temp\\ipykernel_10944\\3439001221.py:4: PerformanceWarning: DataFrame is highly fragmented.  This is usually the result of calling `frame.insert` many times, which has poor performance.  Consider joining all columns at once using pd.concat(axis=1) instead. To get a de-fragmented frame, use `newframe = frame.copy()`\n",
      "  final_df[identifier] = 1\n"
     ]
    }
   ],
   "source": [
    "final_df = pd.concat(all_flattened_runs, ignore_index=True)\n",
    "#final_df = pd.concat([df_only_runs.reset_index(drop=True), final_df], axis=1)\n",
    "final_df = final_df.fillna(0)\n",
    "final_df[identifier] = 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e5f98b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Run</th>\n",
       "      <th>821</th>\n",
       "      <th>46506</th>\n",
       "      <th>562</th>\n",
       "      <th>1755626</th>\n",
       "      <th>28116</th>\n",
       "      <th>46503</th>\n",
       "      <th>39485</th>\n",
       "      <th>817</th>\n",
       "      <th>349096</th>\n",
       "      <th>...</th>\n",
       "      <th>995</th>\n",
       "      <th>999468</th>\n",
       "      <th>36844</th>\n",
       "      <th>69823</th>\n",
       "      <th>76832</th>\n",
       "      <th>1305</th>\n",
       "      <th>58180</th>\n",
       "      <th>467997</th>\n",
       "      <th>652716</th>\n",
       "      <th>Healthy</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SRR7690035</td>\n",
       "      <td>18.960600</td>\n",
       "      <td>14.236300</td>\n",
       "      <td>13.555000</td>\n",
       "      <td>10.40170</td>\n",
       "      <td>6.634150</td>\n",
       "      <td>3.000630</td>\n",
       "      <td>2.807040</td>\n",
       "      <td>2.356580</td>\n",
       "      <td>1.612</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SRR7690037</td>\n",
       "      <td>7.032240</td>\n",
       "      <td>0.391292</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.74015</td>\n",
       "      <td>0.259025</td>\n",
       "      <td>0.347203</td>\n",
       "      <td>0.347203</td>\n",
       "      <td>0.424359</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SRR7690038</td>\n",
       "      <td>18.984600</td>\n",
       "      <td>0.013051</td>\n",
       "      <td>0.047853</td>\n",
       "      <td>8.92678</td>\n",
       "      <td>16.070000</td>\n",
       "      <td>0.796102</td>\n",
       "      <td>0.004350</td>\n",
       "      <td>0.021751</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SRR7690042</td>\n",
       "      <td>15.365700</td>\n",
       "      <td>1.167480</td>\n",
       "      <td>0.005235</td>\n",
       "      <td>7.82158</td>\n",
       "      <td>1.036590</td>\n",
       "      <td>1.649130</td>\n",
       "      <td>1.026120</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SRR7690045</td>\n",
       "      <td>0.130682</td>\n",
       "      <td>6.477270</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>12.95450</td>\n",
       "      <td>4.136360</td>\n",
       "      <td>0.011364</td>\n",
       "      <td>1.335230</td>\n",
       "      <td>1.392050</td>\n",
       "      <td>0.000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows  799 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Run        821      46506        562   1755626      28116     46503  \\\n",
       "0  SRR7690035  18.960600  14.236300  13.555000  10.40170   6.634150  3.000630   \n",
       "1  SRR7690037   7.032240   0.391292   0.000000   6.74015   0.259025  0.347203   \n",
       "2  SRR7690038  18.984600   0.013051   0.047853   8.92678  16.070000  0.796102   \n",
       "3  SRR7690042  15.365700   1.167480   0.005235   7.82158   1.036590  1.649130   \n",
       "4  SRR7690045   0.130682   6.477270   0.000000  12.95450   4.136360  0.011364   \n",
       "\n",
       "      39485       817  349096  ...  995  999468  36844  69823  76832  1305  \\\n",
       "0  2.807040  2.356580   1.612  ...  0.0     0.0    0.0    0.0    0.0   0.0   \n",
       "1  0.347203  0.424359   0.000  ...  0.0     0.0    0.0    0.0    0.0   0.0   \n",
       "2  0.004350  0.021751   0.000  ...  0.0     0.0    0.0    0.0    0.0   0.0   \n",
       "3  1.026120  0.000000   0.000  ...  0.0     0.0    0.0    0.0    0.0   0.0   \n",
       "4  1.335230  1.392050   0.000  ...  0.0     0.0    0.0    0.0    0.0   0.0   \n",
       "\n",
       "   58180  467997  652716  Healthy  \n",
       "0    0.0     0.0     0.0        1  \n",
       "1    0.0     0.0     0.0        1  \n",
       "2    0.0     0.0     0.0        1  \n",
       "3    0.0     0.0     0.0        1  \n",
       "4    0.0     0.0     0.0        1  \n",
       "\n",
       "[5 rows x 799 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0e6d628",
   "metadata": {},
   "outputs": [],
   "source": [
    "final_df.to_csv('Data/Raw_runs/healthy_runs_PRJNA485797', index = False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "base",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
